src/js/stores/MesosStateStore.js

```diff
@@ -194,18 +211,24 @@ class MesosStateStore extends GetSetBaseStore {
   }
 
   getTasksFromNodeID(nodeID) {
-    const services = this.get("lastMesosState").frameworks || [];
+    const frameworks = this.get("lastMesosState").frameworks || [];
     const memberTasks = {};
 
     const schedulerTasks = this.getSchedulerTasks();
 
-    services.forEach(service => {
-      service.tasks.forEach(function(task) {
+    frameworks.forEach(framework => {
+      framework.tasks.forEach(function(task) {
         if (task.slave_id === nodeID) {
-          memberTasks[task.id] = assignSchedulerTaskField(task, schedulerTasks);
+          // Need to get service from Marathon because we need the labels
+          const service = DCOSStore.serviceTree.findServiceByName(
```

Comment:
Since `findServiceByName` is only being used here, only needs the case where we find by `getFrameworkName` and was fundamentally broken previously, I have removed the former behavior in favor of the new behavior.